lib/Twig/Extension/Core.php

```diff
@@ -1259,7 +1259,15 @@ function _twig_escape_html_attr_callback($matches)
      */
     function twig_length_filter(Twig_Environment $env, $thing)
     {
-        return is_scalar($thing) ? mb_strlen($thing, $env->getCharset()) : count($thing);
+        if (is_scalar($thing)) {
+            return mb_strlen($thing, $env->getCharset());
+        }
+
+        if (is_object($thing) && !$thing instanceof \Countable && is_callable(array($thing, '__toString'))) {
```

Comment:
I would use `method_exists()` here instead as `__toString()` must be public and we don't want to return `true` is a class implements `__call()`.